From 00f4f6c7b0776b4f54e5521fe11bcf23d25630ca Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Sat, 22 Nov 2014 23:56:44 -0500
Subject: [md] dm: sparse - Annotate field with __rcu for checking

Message-id: <1416700613-25142-5-git-send-email-snitzer@redhat.com>
Patchwork-id: 100838
O-Subject: [RHEL7.1 PATCH 04/13] dm: sparse: Annotate field with __rcu for checking
Bugzilla: 1165246
RH-Acked-by: Joe Thornber <thornber@redhat.com>

BZ: 1165246

Upstream linux-dm.git commit 6fa9952097747f71c5077f3e14ce3f8adee6f778
Author: Pranith Kumar <bobby.prani@gmail.com>
Date:   Tue Oct 28 15:09:57 2014 -0700

    dm: sparse: Annotate field with __rcu for checking

    Annotate the map field with __rcu since this is a rcu pointer which is checked
    by sparse.

    Signed-off-by: Pranith Kumar <bobby.prani@gmail.com>
    Signed-off-by: Paul E. McKenney <paulmck@linux.vnet.ibm.com>
    Signed-off-by: Mike Snitzer <snitzer@redhat.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/md/dm.c b/drivers/md/dm.c
index 98f12c6..648a686 100644
--- a/drivers/md/dm.c
+++ b/drivers/md/dm.c
@@ -140,7 +140,7 @@ struct mapped_device {
   * Use dm_get_live_table{_fast} or take suspend_lock for
   * dereference.
   */
- struct dm_table *map;
+ struct dm_table __rcu *map;
 
  struct list_head table_devices;
  struct mutex table_devices_lock;
-- 
1.7.1